Can't use Angular CLI with fresh nativescript installation

**Environment**
OS: Ubuntu 20.10 (Groovy Gorilla), 5.8.0-36-generic
node: 14.10.0
nativescript: 7.1.2
package.json: see repo [here](https://github.com/avif/native-test)

**Describe the bug**
I followed the guide to install the Angular CLI here [angular-cli](https://docs.nativescript.org/angular/tooling/angular-cli) with a fresh nativescript installation and could not get the Angualr CLI to work, I also could not successfully downgrade the angular version to get it to work.
On a side note, I also tried downloading the artifact from the playground but it had completely different versions and output js files laying around.

**To Reproduce**
```bash
npm i -g nativescript # nativescript@7.1.2
tns create native-test --ng
cd native-test
npm i -D @angular/cli # @angular/cli@11.1.2
npm i -D @nativescript/schematics # @nativescript/schematics@11.0.0
npx ng generate component test
An unhandled exception occurred: Cannot find module '@schematics/angular/utility/config'
Require stack:
- /home/test/Projects/native-test/node_modules/@nativescript/schematics/src/angular-project-parser.js
- /home/test/Projects/native-test/node_modules/@nativescript/schematics/src/generate/component/index.js
- /home/test/Projects/native-test/node_modules/@angular-devkit/schematics/tools/export-ref.js
- /home/test/Projects/native-test/node_modules/@angular-devkit/schematics/tools/index.js
- /home/test/Projects/native-test/node_modules/@angular/cli/utilities/json-schema.js
- /home/test/Projects/native-test/node_modules/@angular/cli/models/command-runner.js
- /home/test/Projects/native-test/node_modules/@angular/cli/lib/cli/index.js
- /home/test/Projects/native-test/node_modules/@angular/cli/lib/init.js
- /home/test/Projects/native-test/node_modules/@angular/cli/bin/ng
See "/tmp/ng-iQvXCz/angular-errors.log" for further details.
```

**Expected behavior**
A new angular component is created and linked to main module.

**Sample project**
[native-test](https://github.com/avif/native-test)

**Additional context**
I looked at `@nativescript/schematics/src/angular-project-parser.js` and it seems to fail because of changes upstream that moved `getWorkspace` from `@schematics/angular/utility/config` to `@schematics/angular/utility/workspace`, tried to change that and got this error `Cannot read property 'hello-world' of undefined` which suggests it didn't find the workspace.

@Bezlepkin Didn't help..
I tried this:
```
npm i -D @schematics/angular@11.0.0 #(11.0.0) it's version of Angular in my project
npx ng generate component test

Option "entryComponent" is deprecated: Since version 9.0.0 with Ivy, entryComponents is no longer necessary.
Could not find an NgModule. Use the skip-import option to skip importing in NgModule.
```
And with latest version:
```
npm i -D @schematics/angular # 11.2.0
npx ng generate component test

An unhandled exception occurred: Cannot find module '@schematics/angular/utility/config'
Require stack:
- /home/test/Projects/native-test/node_modules/@nativescript/schematics/src/angular-project-parser.js
- /home/test/Projects/native-test/node_modules/@nativescript/schematics/src/generate/component/index.js
- /home/test/Projects/native-test/node_modules/@angular-devkit/schematics/tools/export-ref.js
- /home/test/Projects/native-test/node_modules/@angular-devkit/schematics/tools/index.js
- /home/test/Projects/native-test/node_modules/@angular/cli/utilities/json-schema.js
- /home/test/Projects/native-test/node_modules/@angular/cli/models/command-runner.js
- /home/test/Projects/native-test/node_modules/@angular/cli/lib/cli/index.js
- /home/test/Projects/native-test/node_modules/@angular/cli/lib/init.js
- /home/test/Projects/native-test/node_modules/@angular/cli/bin/ng
See "/tmp/ng-xr1RzJ/angular-errors.log" for further details.
```

--- a/adding-angular-CLI-to-NativeScript.md
+++ b/adding-angular-CLI-to-NativeScript.md
@@ -15,7 +15,12 @@
       "root": "",
       "sourceRoot": ".",
        "projectType": "application",
-       "prefix": "app"
+       "prefix": "app",
+       "schematics": {
+        "@schematics/angular:component": {
+          "style": "scss"
+        }
+       }
     }
   },
   "defaultProject": "my-project-name"

--- a/package.json
+++ b/package.json
@@ -29,11 +29,11 @@
     "@angular-devkit/schematics": "~11.0.0",
     "@nativescript/tslint-rules": "~0.0.5",
     "@phenomnomnominal/tsquery": "^4.1.0",
+    "@schematics/angular": "~11.0.0",
     "strip-json-comments": "~3.1.1"
   },
   "devDependencies": {
     "@angular/cli": "~11.0.0",
-    "@schematics/angular": "~11.0.0",
     "@types/jasmine": "~3.5.0",
     "@types/jasminewd2": "~2.0.3",
     "@types/node": "^12.11.1",

--- a/src/generate/utils.ts
+++ b/src/generate/utils.ts
@@ -1,5 +1,6 @@
-import { getPackageJson } from '../utils'; //getNsConfig
+import { getPackageJson } from '../utils';
 import { Tree, SchematicsException } from '@angular-devkit/schematics';
+import { normalize } from '@angular-devkit/core';
 import { extname } from 'path';
 import { Schema as ComponentOptions } from './component/schema';
 import { Schema as ModuleOptions } from './module/schema';
@@ -31,15 +32,7 @@ const isNs = (tree: Tree) => {
 };
 
 const isWeb = (tree: Tree) => {
-  if (!tree.exists('nativescript.config.ts')) {
-    console.log(`nativescript.config.ts not found. Assuming this is a {N} only project`);
-
-    return false;
-  }
-
-  // const config = getNsConfig(tree);
-
-  return true;//config.webext != null;
+  return tree.exists(normalize('/src/main.tns.ts'));
 };
 
 export interface PlatformUse {
@@ -77,24 +70,16 @@ export const getPlatformUse = (tree: Tree, options: Options): PlatformUse => {
 };
 
 export const getExtensions = (tree: Tree, options: Options): Extensions => {
-  // let ns = options.nsExtension;
-  // let web = options.webExtension;
-
-  // if (isWeb(tree)) {
-  //   const nsconfig = getNsConfig(tree);
-
-  //   ns = ns || nsconfig.nsext;
-  //   web = web || nsconfig.webext;
-
-  //   if (ns === web) {
-  //     ns = DEFAULT_SHARED_EXTENSIONS.ns;
-  //     web = DEFAULT_SHARED_EXTENSIONS.web;
-  //   }
-  // }
+  if (isWeb(tree)) {
+    return {
+      ns: DEFAULT_SHARED_EXTENSIONS.ns,
+      web: DEFAULT_SHARED_EXTENSIONS.web,
+    };
+  }
 
   return {
-    ns: DEFAULT_SHARED_EXTENSIONS.ns,// parseExtension(ns || ''),
-    web: DEFAULT_SHARED_EXTENSIONS.web //parseExtension(web || ''),
+    ns: '',
+    web: '',
   };
 };
 

